frontend/react/style-guide.md

```diff
@@ -91,49 +110,47 @@ src
     });
 
     // good
-    class Listing extends Component {
-      // ...
-      render() {
-        return <div>{this.state.hello}</div>;
-      }
+    function Listing({ hello }) {
+      return <div>{hello}</div>;
     }
     ```
-
-    And if you don't have state or refs, only for stateless components, prefer normal functions (not arrow functions) over classes:
+  - Prefer normal function components (not arrow functions) over `class extends Component`. Only exception is usage of `componentDidCatch` and `getDerivedStateFromError` (which have no hooks support)
 
     ```jsx
     // bad
     class Listing extends Component {
+      state = { foo: 1 };
+
       render() {
-        return <div>{this.props.hello}</div>;
+        return <div>{this.state.hello}</div>;
       }
     }
 
-    // bad (relying on function name inference is discouraged)
-    const Listing = ({ hello }) => (
-      <div>{hello}</div>
-    );
+    // bad
+    const Listing = ({ hello }) => {
+      const [foo, setFoo] = useState(1);
+      return <div>{hello}</div>;
+    }
 
     // good
     function Listing({ hello }) {
+      const [foo, setFoo] = useState(1);
```

Comment:
I'm not sure about encouraging the use of hooks yet. In my case I tell trainees and devs who i review to have this guide handy for consulting and it may be confusing for them as not everyone knows about them and it is not even included in the training.
I'd include this later when we have a more mature standard.

Reply:
Hmm I haven't though it that yet... But I'd like that all new trainees follow this and all new projects too. So we should make sure the training does. Since we have just changed it/are changing it, I think it's a good chance to ensure this happens. @pabloccid what do you think?

Reply:
If we plan to extend the use of hooks to trainings and teach other devs about them then it's ok to me. 
👍

Reply:
I'm abord on using hooks instead of classes, but i think we should have in mind to create an express training for hooks.

Reply:
I think it's a good idea 👍